packages/frontend/apps/electron/src/main/recording/index.ts

```diff
@@ -1,35 +1,31 @@
-import { ShareableContent, TappableApplication } from '@affine/native';
-import { Notification } from 'electron';
-import {
-  BehaviorSubject,
-  distinctUntilChanged,
-  pairwise,
-  startWith,
-} from 'rxjs';
+import path from 'node:path';
+
+import { ShareableContent } from '@affine/native';
+import { app, nativeImage, Notification } from 'electron';
+import fs from 'fs-extra';
+import { debounce } from 'lodash-es';
+import { BehaviorSubject, distinctUntilChanged, groupBy, mergeMap } from 'rxjs';
 
 import { isMacOS } from '../../shared/utils';
 import { beforeAppQuit } from '../cleanup';
+import { ensureHelperProcess } from '../helper-process';
 import { logger } from '../logger';
-
-interface TappableAppInfo {
-  rawInstance: TappableApplication;
-  isRunning: boolean;
-  processId: number;
-  processGroupId: number;
-  bundleIdentifier: string;
-  name: string;
-}
-
-interface AppGroupInfo {
-  processGroupId: number;
-  apps: TappableAppInfo[];
-  name: string;
-  icon: Buffer | undefined;
-  isRunning: boolean;
-}
+import type { NamespaceHandlers } from '../type';
+import { getMainWindow } from '../windows-manager';
+import type {
+  AppGroupInfo,
+  Recording,
+  RecordingStatus,
+  TappableAppInfo,
+} from './types';
 
 const subscribers: Subscriber[] = [];
 
+const SAVED_RECORDINGS_DIR = path.join(
```

Comment:
why do we save recording in fs?

Reply:
it only make sense if we streaming the recording to file, in case the program crash somehow, or randomly quit, to prevent data of recorded audios

Reply:
and we should clear them after them've been put in pages, to save disks